blockchain/chain_sync/src/sync.rs

```diff
@@ -620,30 +632,44 @@ where
         let db = Arc::clone(&self.chain_store.db);
         let parent_clone = parent_tipset.clone();
         // check messages to ensure valid state transitions
-        let x = task::spawn_blocking(move || Self::check_block_msgs(db, b, &parent_clone));
+        let sm = self.state_manager.clone();
+        let x = Self::check_block_msgs(sm, db, b, parent_clone).boxed();
         validations.push(x);
 
-        // TODO use computed state_root instead of parent_tipset.parent_state()
-
         // block signature check
+        let (state_root, _) = self
+            .state_manager
+            .write()
+            .await
+            .tipset_state(&parent_tipset)
+            .map_err(|_| Error::Validation("Could not update state".to_owned()))?;
         let work_addr_result = self
             .state_manager
-            .get_miner_work_addr(&parent_tipset.parent_state(), header.miner_address());
+            .read()
+            .await
+            .get_miner_work_addr(&state_root, header.miner_address());
+
+        // temp header needs to live long enough
         match work_addr_result {
-            Ok(work_addr) => {
-                let temp_header = header.clone();
-                let block_sig_task = task::spawn_blocking(move || {
-                    temp_header
-                        .check_block_signature(&work_addr)
-                        .map_err(Error::Blockchain)
-                });
-                validations.push(block_sig_task)
+            Ok(_) => {
+  
+                validations.push(
+                    async {
+                        block
+                            .header()
+                            .check_block_signature(work_addr_result.unwrap().clone()) //work_addr_result lives longer than unwrapped_value in scope
```

Comment:
can you remove the comment? unclear what exactly the meaning is

(Here is where you can add the reference wrt the previous comment and it works fine)

Reply:
The unwrapped Ok in this instance won't live as long as this work_result_addr and that is why I have to unwrap

Reply:
I don't have a problem with that